src/components/PeoplePage.tsx

```diff
@@ -1,35 +1,116 @@
+import React, {
+  memo, useEffect, useMemo, useState,
+} from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { getVisiblePeople } from '../utils/getVisiblePeople';
+
+export const PeoplePage: React.FC = memo(() => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const [searchParams] = useSearchParams();
+
+  const query = searchParams.get('query');
+  const sex = searchParams.get('sex');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+  const century = searchParams.getAll('century');
+
+  const setLoadedPeople = async () => {
+    setIsLoading(true);
+
+    try {
+      setIsLoading(true);
+      const loadPeople = await getPeople();
+
+      const peopleWithParents = loadPeople.map(person => {
+        const father = loadPeople.find(f => f.name === person.fatherName);
+        const mother = loadPeople.find(m => m.name === person.motherName);
+
+        return (
+          {
+            ...person,
+            father,
+            mother,
+          }
+        );
+      });
+
+      setPeople(peopleWithParents);
+    } catch (error) {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    setLoadedPeople();
+  }, []);
+
+  const filteredPeople = useMemo(() => getVisiblePeople({
+    people,
+    query,
+    sex,
+    century,
+    sort,
+    order,
+  }), [
+    people,
+    query,
+    sex,
+    sort,
+    century,
+    order,
+  ]);
```

Comment:
I suggest you to create a custom hook `useVisiblePeople` which takes an object with params and return `useMemo` result of filtering and sorting
```
export const useVisiblePeople = (params) => {
  return useMemo(() => {
   // filter logic
}, [bla])
}